lib/identity_cache/query_api.rb

```diff
@@ -476,5 +484,15 @@ def was_new_record? # :nodoc:
       pk = self.class.primary_key
       !destroyed? && transaction_changed_attributes.has_key?(pk) && transaction_changed_attributes[pk].nil?
     end
+
+    def load_and_readonlyify(association_name)
+      record_or_records = send(association_name)
+      if self.class.reflect_on_association(association_name).collection?
+        record_or_records.map(&:readonly!)
+      else
+        record_or_records.try(:readonly!)
```

Comment:
@dylanahsmith 
> so it would be safe to modify and save it

Are you referring to the record from the cache here or the active record association?